packages/tizen_rpc_port/tizen/src/rpc_port_proxy.cc

```diff
@@ -0,0 +1,223 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "rpc_port_proxy.hh"
+
+#include <bundle.h>
+#include <flutter/standard_message_codec.h>
+#include <rpc-port-parcel.h>
+
+#include <cstdint>
+#include <utility>
+#include <vector>
+
+#include "log.h"
+
+namespace tizen {
+
+namespace {
+
+flutter::EncodableMap CreateEncodableMap(const char* event,
+                                         const char* receiver,
+                                         const char* port_name) {
+  return {{flutter::EncodableValue("event"),
+           flutter::EncodableValue(std::string(event))},
+          {flutter::EncodableValue("receiver"),
+           flutter::EncodableValue(std::string(receiver))},
+          {flutter::EncodableValue("portName"),
+           flutter::EncodableValue(std::string(port_name))}};
+}
+
+}  // namespace
+
+RpcPortProxy::RpcPortProxy(std::string appid, std::string port_name)
+    : appid_(std::move(appid)), port_name_(std::move(port_name)) {
+  LOG_DEBUG("RpcPortProxy: %s/%s", appid_.c_str(), port_name_.c_str());
+  int ret = rpc_port_proxy_create(&handle_);
+  if (ret != RPC_PORT_ERROR_NONE)
+    LOG_ERROR("rpc_port_proxy_create() is failed. error: %d", ret);
+}
+
+RpcPortProxy::~RpcPortProxy() {
+  if (handle_ != nullptr) rpc_port_proxy_destroy(handle_);
+}
+
+RpcPortResult RpcPortProxy::Connect(EventSink sink) {
+  LOG_DEBUG("Connect: %s/%s", appid_.c_str(), port_name_.c_str());
+  if (connected_) {
+    LOG_ERROR("Already connected");
+    return CreateResult(RPC_PORT_ERROR_INVALID_PARAMETER);
+  }
+
+  event_sink_ = std::move(sink);
+  int ret =
+      rpc_port_proxy_add_connected_event_cb(handle_, OnConnectedEvent, this);
+  if (ret != RPC_PORT_ERROR_NONE) {
+    return CreateResult(ret);
+  }
+
+  ret = rpc_port_proxy_add_disconnected_event_cb(handle_, OnDisconnectedEvent,
+                                                 this);
+  if (ret != RPC_PORT_ERROR_NONE) {
+    return CreateResult(ret);
+  }
+
+  ret = rpc_port_proxy_add_rejected_event_cb(handle_, OnRejectedEvent, this);
+  if (ret != RPC_PORT_ERROR_NONE) {
+    return CreateResult(ret);
+  }
+
+  ret = rpc_port_proxy_add_received_event_cb(handle_, OnReceivedEvent, this);
+  if (ret != RPC_PORT_ERROR_NONE) {
+    return CreateResult(ret);
+  }
+
+  ret = rpc_port_proxy_connect(handle_, appid_.c_str(), port_name_.c_str());
+  if (ret != RPC_PORT_ERROR_NONE) {
+    return CreateResult(ret);
+  }
+
+  connected_ = true;
```

Comment:
Is `rpc_port_proxy_connect` asynchronous? Then `connected_` should be updated in the callback.

Reply:
It is just checks whether that connect was called.
Um.. I think this flag may be not needed. I removed it.